actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -497,6 +508,17 @@ def expiry_options(options)
           end
         end
 
+        def cookie_metadata(options)
+          if request.use_cookies_with_metadata
+            metadata = expiry_options(options)
+            metadata[:purpose] = options[:cookie_name]
+          else
+            metadata = {}
```

Comment:
Could skip the assignment here, return the literal straight and move the return below up to the if branch :)